app/src/main/contrib/components/Responsibilities/ModelRunParameters/ModelRunParametersStatus.tsx

```diff
@@ -19,41 +25,73 @@ export class ModelRunParametersStatus extends React.Component<Props, State> {
         super();
         const storeState = runParametersStore.getState();
         this.state = {
-            sets: storeState.parameterSets
-                ? storeState.parameterSets.filter(s => s.disease == props.disease)
-                : []
+            set: this.getLastSetForDisease(storeState.parameterSets, props.disease),
+            tokens: storeState.oneTimeTokens,
+            groupId: storeState.groupId,
+            touchstoneId: storeState.touchstoneId
         }
+        this.refreshToken = this.refreshToken.bind(this)
+    }
+
+    getLastSetForDisease(allSets: ModelRunParameterSet[], disease: string) {
+        const sets = allSets
+            ? allSets.filter(s => s.disease == disease)
+            : []
+        if (sets.length) {
+            return sets[0]
+        }
+        return null;
     }
 
     componentDidMount() {
         runParametersStore.listen(this.onChange.bind(this));
+        if (this.state.set) {
+            runParameterActions.fetchToken(this.state.groupId, this.state.touchstoneId, this.state.set.id)
+        }
     }
 
     onChange(storeState: RunParametersState) {
+        const oldSetId = this.state.set ? this.state.set.id : null;
+        this.setState({
+            set: this.getLastSetForDisease(storeState.parameterSets, this.props.disease),
+            tokens: storeState.oneTimeTokens
+        })
+        const newSetId = this.state.set ? this.state.set.id : null;
+        if (newSetId && oldSetId != newSetId) {
+            runParameterActions.fetchToken(storeState.groupId, storeState.touchstoneId, newSetId)
+        }
+    }
 
-        if (storeState.parameterSets) {
-            this.setState({sets: storeState.parameterSets.filter(s => s.disease == this.props.disease)})
+    refreshToken(setId: number) {
+        return () => {
+            runParameterActions.clearToken(setId);
+            const s = runParametersStore.getState();
```

Comment:
can get rid of this call to `getState` and use the `groupId` and `touchstoneId` from local state now